File: test/Data.Analyzers.Vsix.Test/CSharp/PrimaryKeyArgumentNaming.cs
namespace DevZest.Data.Analyzers.Vsix.Test.CSharp
{
    public class PrimaryKeyArgumentNaming : Model<PrimaryKeyArgumentNaming.PK>
    {
        public sealed class PK : CandidateKey
        {
            public PK(_Int32 id2)
                : base(id2)
            {
            }
        }

        protected sealed override PK CreatePrimaryKey()
        {
            return new PK(ID);
        }

        public static readonly Mounter<_Int32> _ID = RegisterColumn((PrimaryKeyArgumentNaming _) => _.ID);

        public _Int32 ID { get; private set; }
    }
}

File: test/Data.Analyzers.Vsix.Test/VisualBasic/PrimaryKeyArgumentNaming.vb
Public Class PrimaryKeyArgumentNaming
    Inherits Model(Of PrimaryKeyArgumentNaming.PK)

    Public NotInheritable Class PK
        Inherits PrimaryKey
        Public Sub New(id2 As _Int32)
            MyBase.New(id2)
        End Sub
    End Class

    Protected NotOverridable Overrides Function CreatePrimaryKey() As PK
        Return New PK(ID)
    End Function

    Public Shared ReadOnly _ID As Mounter(Of _Int32) = RegisterColumn(Function(x As PrimaryKeyArgumentNaming) x.ID)

    Private m_ID As _Int32
    Public Property ID As _Int32
        Get
            Return m_ID
        End Get
        Private Set
            m_ID = Value
        End Set
    End Property
End Class
